src/Transition/CreateNewIDFUsingRulesV8_8_0.f90

```diff
@@ -366,6 +389,66 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
                  OutArgs(15:CurArgs+1)=InArgs(14:CurArgs)  !
                  CurArgs = CurArgs + 1
 
+             CASE ('BUILDINGSURFACE:DETAILED')
+                 CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                 nodiff=.false.
+                 IF ( SameString(InArgs(5), 'Foundation' ) .and. SameString(InArgs(2), 'Floor' ) ) THEN
+                   NumPerimObjs = GetNumObjectsFound('SURFACEPROPERTY:EXPOSEDFOUNDATIONPERIMETER')
+                   PerimNum = 0
+                   DO xcount=1,NumPerimObjs
+                     CALL GetObjectItem('SURFACEPROPERTY:EXPOSEDFOUNDATIONPERIMETER',xcount,Alphas,NumAlphas,Numbers,NumNumbers,Status)
+                     IF ( SameString( TRIM(Alphas(1)), TRIM(InArgs(1)) ) ) THEN
+                       PerimNum = xcount
```

Comment:
Fair points. It might be helpful for debugging to know what the object number is.

It's also been a while since I've fortraned. Do you mean `EXIT`?

```
                   DO xcount=1,NumPerimObjs
                     CALL GetObjectItem('SURFACEPROPERTY:EXPOSEDFOUNDATIONPERIMETER',xcount,Alphas,NumAlphas,Numbers,NumNumbers,Status)
                     IF ( SameString( TRIM(Alphas(1)), TRIM(InArgs(1)) ) ) THEN
                       PerimNum = xcount
                       EXIT
                     ENDIF
                   ENDDO

```